### DISCUSSION TITLE
EEPROM Size 512 bytes but only 256 bytes read/write

### DISCUSSION THREAD

**Discussion by Zampano290** [SpenceKonde/DxCore#219] [Q&A] [answered]
I have build a device with a mega4809. This controller only have 256byte eeprom, but i need 512bytes for my device, so i need a external eeprom.
yesterday i test now the avr128db48 (100% pin comp. to the 4809) on the pcb without external eeprom.

all work perfect, but only the eeprom work wrong.
eeprom.length() get return 512
when i write a config block to address 256-511 and then read from 0-511 i get always the same values from 0-255 an 256-511

**Comment by SpenceKonde** [comment]
Thanks for reporting, yeah looks like it was introduced while harmonizing the EEPROM libraries on DxCore and megaTinyCore. If the size of the EEPROM is 256 or less, you want to represent the index with a uint8_t, especially on a part that might only have 4k of flash total.

**Comment by Zampano290** [comment]
Thanks for Fix the Library, but with changes it does´n run.
I modify the EEPROM.h to test on my board, i have change uint8_t index to uint16_t index, now I can access the EEPROM >255 bytes.
[EEPROM.txt](https://github.com/SpenceKonde/DxCore/files/7806607/EEPROM.txt)

**Comment by SpenceKonde** [comment]
Yeah, changing the index is exactly the fix I implemented. 
The fix is not in the board manager release because I need to fix additional critical bugs before releasing.

**Reply by SpenceKonde** [reply]
okay that fix actuall wasn't complete, completed fix is checked in will do release tomorrow.

**Comment by SpenceKonde** [comment]
Just want to say that this is now resolved. as far as I know.

**Comment by Zampano290** [comment, marked as answer]
Thank you, I have installed the newest version today and test on my actual project. Work fine!
